### PR TITLE
DOC-1352: Documenting auto* relationships

### PR DESCRIPTION
**Pull request by jtappa**

### CODE REVIEW

File: source/puppet/4.2/reference/lang_relationships.markdown [hunk 1]
```diff
@@ -220,13 +220,11 @@ Only certain resource types can refresh themselves. Of the built-in resource typ
 
 Service resources refresh by restarting their service. Mount resources refresh by unmounting and then mounting their volume. Exec resources usually do not refresh, but can be made to: setting `refreshonly => true` causes the exec to never fire _unless_ it receives a refresh event. You can also set an additional `refresh` command, which causes the exec to run both commands when it receives a refresh event.
 
-### Autorequire
+### Auto* Relationships
```
Comment (nfagerlund) [inline source/puppet/4.2/reference/lang_relationships.markdown]: Backslash-escape the \*, because Markdown. :|

File: source/puppet/4.2/reference/lang_relationships.markdown [hunk 2]
```diff
@@ -220,13 +220,11 @@ Only certain resource types can refresh themselves. Of the built-in resource typ
...
 
-When Puppet is preparing to sync a resource whose type supports autorequire, it will search the catalog for any resources that match certain rules. If it finds any, it will process them _before_ that resource. If Puppet _doesn't_ find any resources that could be autorequired, that's fine; they won't be considered a failed dependency.
-
-The [resource type reference][type] contains information on which resource types can autorequire other resources. Each resource type's description should state its autorequire behavior, if any. For an example, see the "Autorequires" section near the end of [the `exec` type's description.][exec]
+When Puppet is preparing to sync a resource whose type supports an auto relationship, it will search the catalog for any resources that match certain rules. If it finds any, it will process them _before_ that resource. If Puppet _doesn't_ find any resources that could use an auto relationship, that's fine; they won't be considered a failed dependency.
```
Comment (nfagerlund) [inline source/puppet/4.2/reference/lang_relationships.markdown]: Well, if it's an autobefore it will process them _after_ that resource. I'd reword to say "if it finds any, it will process them in the correct order, sending refresh events if necessary."

**Comment by nfagerlund** [comment]
Two tweaks; other than that, lgtm.

File: source/puppet/4.2/reference/lang_relationships.markdown (hunk 1 above)
Comment (jtappa) [inline source/puppet/4.2/reference/lang_relationships.markdown]: (╯°□°）╯︵ ┻━┻)

Comment (nfagerlund) [inline source/puppet/4.2/reference/lang_relationships.markdown]: p. much

**Comment by jtappa** [comment]
@nfagerlund pushed tweaks!
